### DISCUSSION TITLE
includeBuild 'kotlin-toolkit'

### DISCUSSION THREAD

**Discussion by Frankwmf** [readium/kotlin-toolkit#498] [Q&A]
i added includeBuild 'kotlin-toolkit' in settings.gradle. Is it need to add implementation in build.gradle? i try implementation project(':kotlin-toolkit') or implementation ':kotlin-toolkit' , all are fail.  How to use it ? Many Thanks!

**Comment by mickael-menu** [comment]
Are you using a Git submodule for Readium? If not, you should not use `includeBuild 'kotlin-toolkit'`.

Even with a submodule, you need to add the dependencies to your project with the full name:

```gradle
dependencies {
    implementation "org.readium.kotlin-toolkit:readium-shared:$readium_version"
    implementation "org.readium.kotlin-toolkit:readium-streamer:$readium_version"
    implementation "org.readium.kotlin-toolkit:readium-navigator:$readium_version"
    implementation "org.readium.kotlin-toolkit:readium-opds:$readium_version"
    implementation "org.readium.kotlin-toolkit:readium-lcp:$readium_version"
}
```

**Reply by Frankwmf** [reply]
Thank you very much!

**Comment by Frankwmf** [comment]
|--my-app
	|--app
		| --build.gradle
		| --src/main/java
	| settings.gradle >>  (includeBuild 'kotlin-toolkin')
|--kotlin-toolkin
	|--Readium
	|--test-app
	

Dear Sir, is it possible to call test-app's BookshelfViewModel to open epub in my-app? e.g. my-app/app/main_activity.java , Thank you!

**Reply by mickael-menu** [reply]
Don't use (or copy-paste whole portions of) the Test App directly in your app. It's just an example showing how to use the Readium toolkit.
